support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt

```diff
@@ -57,29 +55,27 @@ class SendConversation @Inject constructor(
     }
 
     private suspend fun findConversation(parameters: Map<String, Any>): Conversation? {
+        val refId = parameters[PARAM_REF_ID] as Int
         //load page 0
         var page = 0
         var conversationsResult = restApi.fetchConversations(page)
         var conversations = conversationsResult.conversations
-        var totalCount = conversationsResult.totalCount
-        var countLoaded = conversations?.size ?: 0
+        val pageCount = conversationsResult.pageCount
 
-        val refId = parameters[PARAM_REF_ID] as Int
         var conversation = findConversation(conversations, refId)
         if (conversation != null) {
             return conversation
         } else {
-            //fetch again
-            while (totalCount < countLoaded) {
-                page ++
+            //fetch again until
+            page++
+            while (page < pageCount) {
                 conversationsResult = restApi.fetchConversations(page)
                 conversations = conversationsResult.conversations
-                totalCount += conversationsResult.totalCount
-                countLoaded = conversations?.size ?: 0
                 conversation = findConversation(conversations, refId)
                 if (conversation != null) {
                     return conversation
                 }
+                page++
```

Comment:
Here maybe you want to do something like a `do  { ... } while ()`?

Reply:
to make it simpler to read..

Reply:
done!